src/Middleware/WorkerMiddleware.php

```diff
@@ -141,4 +145,33 @@ private function getMiddlewaresForMessage(string $messageName): array
 
         return $mappedMiddlewares;
     }
+
+    /**
+     * @param ServerRequestInterface $request
+     */
+    private function assertLocalhost(ServerRequestInterface $request)
+    {
+        $serverParams = $request->getServerParams();
+        $remoteAddr   = $serverParams['REMOTE_ADDR'] ?? 'unknown IP address';
+
+        // If request is not originating from localhost or from Docker local IP, we throw an RuntimeException
+        if (!in_array($remoteAddr, self::LOCALHOST_ADDRESSES) && !fnmatch('172.17.*', $remoteAddr)) {
+            throw new RuntimeException(sprintf(
+                'Worker requests must come from localhost, request originated from %s given',
+                $remoteAddr
+            ));
+        }
+    }
+
+    /**
+     * @param ServerRequestInterface $request
+     */
+    private function assertSqsUserAgent(ServerRequestInterface $request)
+    {
+        $userAgent = $request->getHeaderLine('User-Agent');
```

Comment:
I was just referring to the fact that if AWS decide to change their UserAgent to something like "Aws-Sqsd" for isntance, in which case your comparaison would fail. Or you could simply use "stripos" instead to make the comparaison case-insensitive.